dockstore/dockstore-ui2
# Fix missing unpublished workflows on refresh

**Description**
The unpublished tab was empty if an unpublished workflow was selected and the browser was refreshed because the tab group was quickly selecting the published tab then the unpublished tab. This seems to be a problem with the animation when the tab content changes quickly (found this issue https://github.com/angular/components/issues/25236). 

It selected the published tab first because the `workflowId` observable that the `selectTab` pipe was waiting for was not ready yet so the pipe selected the published tab according to its logic. Then when `workflowId` was ready, the pipe selected the unpublished tab, resulting in the empty tab. This PR adds a condition to the mat-tab-group so that the tabs are only rendered if the `workflowId` is ready. This ensures that the correct tab is always selected on the first render. While I was in there, I did some minor simplifications of the select tab pipe's logic.

I also fixed an `undefined` console error I noticed in the My Tools page when I tried to reproduce this bug for GitHub app tools.

**Review Instructions**
On QA:
- Go to My Workflows
  - Select an unpublished workflow
  - Refresh the page
  - The unpublished tab that your workflow belongs to should not be empty
- Go to My Tools
  - Select an unpublished GitHub App tool
  - Refresh the page
  - The unpublished tab that your workflow belongs to should not be empty
  - Check that the tabs of the other expansion panels look okay (they're not empty)

**Issue**
dockstore/dockstore#4882
[DOCK-2158](https://ucsc-cgl.atlassian.net/browse/DOCK-2158)

**Security**
If there are any concerns that require extra attention from the security team, highlight them here.

Please make sure that you've checked the following before submitting your pull request. Thanks!
